src/refcount.rs

```diff
@@ -113,9 +122,16 @@ impl RefCounter for Weak {
         Either::Weak(self)
     }
 
-    fn as_ptr(&self) -> *const AtomicBool {
+    fn as_ptr(&self) -> *const (AtomicBool, DropNotice) {
         ArcWeak::as_ptr(&self.0)
     }
+
+    fn disconnect_notice(&self) -> DropNotice {
+        match self.0.upgrade() {
+            Some(arc) =>  arc.1.clone(),
```

Comment:
nit: unnecessary space here. It's probably just best to run the whole PR through rustfmt